minigunapp/tasks.py

```diff
@@ -0,0 +1,40 @@
+import logging
+
+from celery import shared_task
+from minigunapp.models import Email
+from django.core.mail import send_mail
+
+
+@shared_task
+def send_email(id):
+    """ Task that attempts to send the email with the given id.
+    Retries on a delay until max failures reached.
+    """
+
+    email = Email.objects.get(id=id)
```

Comment:
It throws an exception, and the task fails. Celery will try the task 3 more times on the specified delay.